config/app/com/microsoft/azure/iotsolutions/uiconfig/services/external/UserManagementClient.java

```diff
@@ -0,0 +1,89 @@
+
+import com.google.inject.Inject;
+import com.microsoft.azure.iotsolutions.uiconfig.services.exceptions.ExternalDependencyException;
+import com.microsoft.azure.iotsolutions.uiconfig.services.exceptions.NotAuthorizedException;
+import com.microsoft.azure.iotsolutions.uiconfig.services.runtime.IServicesConfig;
+import org.apache.http.HttpStatus;
+import play.Logger;
+import play.libs.Json;
+import play.libs.ws.WSClient;
+import play.mvc.Http;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.CompletionException;
+import java.util.concurrent.CompletionStage;
+
+public class UserManagementClient implements IUserManagementClient {
+
+    private static final String DEFAULT_USER_ID = "default";
+    private static final Logger.ALogger log = Logger.of(UserManagementClient.class);
+
+    private final WSClient wsClient;
+    private final String userManagementServiceUrl;
+
+    @Inject
+    public UserManagementClient(final IServicesConfig config, final WSClient wsClient) {
+        this.wsClient = wsClient;
+        this.userManagementServiceUrl = config.getUserManagementApiUrl();
+    }
+
+    @Override
+    public CompletionStage<List<String>> getAllowedActionsAsync(String userObjectId, List<String> roles) {
+        String url = String.format("%s/users/%s/allowedActions", this.userManagementServiceUrl, userObjectId);
+        if (roles == null) {
+            roles = new ArrayList<>();
+        }
+
+        return this.wsClient.url(url)
+                .post(Json.toJson(roles))
+                .handle((response, error) -> {
+                    if (error != null) {
+                        String message = String.format("Failed to get allowed actions: %s", url);
+                        log.error(message, error.getCause());
+                        throw new CompletionException(message, error.getCause());
+                    } else if (response.getStatus() != Http.Status.OK) {
+                        String message = String.format("Failed to get allowed actions: %s", url);
+                        log.error(message);
+                        throw new CompletionException(new ExternalDependencyException(message));
+                    } else {
+                        return Json.fromJson(response.asJson(), List.class);
+                    }
+                });
+    }
+
+    @Override
+    public CompletionStage<String> getTokenAsync() throws CompletionException {
+        String url = String.format("%s/users/%s/token", this.userManagementServiceUrl, DEFAULT_USER_ID);
+
+        return this.wsClient.url(url)
+                .get()
+                .handle((response, error) -> {
+                    if (error != null) {
+                        // If the error is 403, the user who did the deployment is not authorized
+                        // to assign the role for the application to have contributor access.
+                        if (response.getStatus() == HttpStatus.SC_FORBIDDEN ||
```

Comment:
is there a way to reduce the duplicated code between this and AzureResourceManagerClient?

Reply:
Added a WsResponseHelper class